added option to add headers

tonyeung opened the pull request:
Not sure why I had to call options twice
USAGE:

```
            options: {
                withCredentials: true,
                headers: [
                    {
                        name: 'Authorization',
                        value: 'Bearer ' + localStorageService.get('bearer')
                    }
                ]
            },
```

tonyeung committed:
added option to add headers

USAGE:


            options: {
                withCredentials: true,
                headers: [
                    {
                        name: 'Authorization',
                        value: 'Bearer ' + localStorageService.get('bearer')
                    }
                ]
            },

PowerKiKi commented:
I think it would be best to have headers in a object with key/value, instead of array of objects. As seen in [Angular core's $http](https://docs.angularjs.org/api/ng/service/$http#setting-http-headers)

tonyeung commented:
I'll modify it tomorrow when I get to work.

tonyeung committed:
changed header format from array to keyvalue

usage:

```
            options: {
                withCredentials: true,
                headers: { Authorization: 'Bearer ' + localStorageService.get('bearer')}
            },
```

PowerKiKi commented:
Done in #19
